pandas/core/computation/eval.py

```diff
@@ -232,8 +237,13 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+    modifiable = hasattr(target, "__setitem__")
+
+    if inplace and not modifiable:
+        raise ValueError("Cannot modify the provided target inplace")
```

Comment:
Because now you could pass a target, and then set inplace=True (then this error will not be triggered), while this makes no sense when there is no assignment